modules/ROOT/pages/fd-scatter-gather.adoc

```diff
@@ -0,0 +1,259 @@
+Delena Hodgdon,0028t000001J3OSQ2,hodgdond@greelo.com
+Britt Lafferty,0028t000001J6HUT7,britt.lafferty@hoipolloi.com
+Irving Bates,0028t000001J3IUe6,irving.bates@hoipolloi.com
+Merle Cater,0028t000001J3PPl0,merle.cater@hoipolloi.com
+----
+
+== Saving Message Data into Variables
+
+If you want a route to act on only part of a message, you can store what you need for the route in a variable before the message reaches the Scatter-Gather card or at the beginning of the route.
+
+This configuration is useful if your variables need to be used in more than one route:
+
+.You can save data in one or more variables that one or more routes can use.
+image::fd-sg-alt-1.png[Saving data for use by more than one route]
+
+In this case, the first card in each route receives the same message, which might have these parts if you create two variables before the Scatter-Gather card:
+
+----
+payload
+attributes
+vars
+  var1
+  var2
+----
+
+You can use both variables in both routes, if you need to.
+
+The following configuration is useful if your variables need to be used only in single routes:
+
+.You can save data in a variable that is used by only one route.
+image::fd-sg-alt-2.png[Saving data for use by one route only]
+
+Suppose that you create `var1` at the start of the first route, and you create `var2` at the start of the second route. Then, the second card in the first route receives a message that has these parts:
+
+----
+payload
+attributes
+vars
+  var1
+----
+
+And the second card in the second route receives a message that has these parts:
+
+----
+payload
+attributes
+vars
+  var2
+----
+
+In both of the two previous diagrams, the final card could be a Transform card, which might require you to know how to write scripts in DataWeave, or another Set Variable card that, as in the example about aggregating contacts, combines somehow the results from the routes in the Scatter-Gather card and then passes the variable to another card, such as a Transform card.
+
+== Differences between Parallel For-Each, For-Each, and Scatter-Gather
+
+For-Each iterates through a collection of elements, processing one element at a time through a single route.
+
+Parallel For-Each iterates though a collection of elements _n_ times (where _n_ <= the number of elements in the collection) with a single process. For example, if there are 10 elements and the maximum number of processes is set to 2, a Parallel For-Each card processes two elements at a time, five times.
+
+Scatter Gather runs two or more parallel processes, called _routes_. When all routes are finished, the results from each route are gathered together into a new payload.
+
+== Fields in the Scatter-Gather Card
+
+Timeout:: The maximum length of time to wait, in milliseconds, for the results from each of the parallel processes to be gathered together into a new payload and sent to the next card. A value of 0 or lower means that there is no timeout. The default value is 9,223,372,036,854,776,000 milliseconds.
+
```

Comment:
I would avoid putting the default here.
That value btw is MAX of 64-bit floats. I believe users are not that much interested in that bit, but indicating that it's a 64-bit float may be useful.

Reply:
@matibenav , the problem is that the default appears in the UI. Users will wonder why.